### PR TITLE
NEP: Add paragraph to NEP 41 about no array-object use and fix links

### PR DESCRIPTION
**Pull request by seberg**
This adds a paragraph to the effect of what Travis Oliphant was discussing on the mailing list. To add the links between the documents (without spelling out the full document name), I had to modify the script a bit.

Could be merged as is, or we wait on further changes and feedback. While I guess Eric is still thinking on it, and may have larger comments, I will still take this chance to prod Nathaniel and Stephan since I would love for a quick response on the NEP, @njsmith and @shoyer.

### CODE REVIEW

File: doc/neps/tools/build_index.py [hunk 1]
```diff
@@ -34,15 +34,23 @@ def nep_metadata():
...
+
+            tags['Title'] = lines[i+1].strip()
             tags['Filename'] = source
 
         if not tags['Title'].startswith(f'NEP {nr} — '):
             raise RuntimeError(
                 f'Title for NEP {nr} does not start with "NEP {nr} — " '
-                '(note that — here is a special, enlongated dash)')
+                '(note that — here is a special, enlongated dash). Got: '
+                '    {tags["Title"]}')
```
Comment (seberg) [inline doc/neps/tools/build_index.py]:
```suggestion
               f '    "{tags["Title"]}"')
```

Comment (seberg) [reply doc/neps/tools/build_index.py]: OK, fixed this up, and debugged the error branches (this and new one) manually...

File: doc/neps/nep-0041-improved-dtype-support.rst [hunk 2]
```diff
@@ -156,6 +159,13 @@ Further, the public API will be designed in a way that is extensible in the futu
    as much as possible. The public API should be an identical, but limited,
    version of the C-API used for the internal NumPy datatypes.
 
+The datatype system may be targeted to work with NumPy arrays,
+for example by providing strided-loops, but should avoid using the array-object.
```
Comment (eric-wieser) [inline doc/neps/nep-0041-improved-dtype-support.rst]: Not sure if this clarification is needed, but...
```suggestion
for example by providing strided-loops, but should avoid direct interactions with the array-object (typically `np.ndarray` instances).
```

File: doc/neps/tools/build_index.py [hunk 3]
```diff
@@ -34,15 +34,23 @@ def nep_metadata():
...
+
+            tags['Title'] = lines[i+1].strip()
             tags['Filename'] = source
 
         if not tags['Title'].startswith(f'NEP {nr} — '):
             raise RuntimeError(
                 f'Title for NEP {nr} does not start with "NEP {nr} — " '
-                '(note that — here is a special, enlongated dash)')
+                '(note that — here is a special, enlongated dash). Got: '
+                f'    "{tags["Title"]}"')
```
Comment (eric-wieser) [inline doc/neps/tools/build_index.py]:
```suggestion
                f'    {tags["Title"]!r}')
```

File: doc/neps/nep-0041-improved-dtype-support.rst (hunk 2 above)
Comment (seberg) [reply doc/neps/nep-0041-improved-dtype-support.rst]: Can't hurt, thanks!

**Comment by seberg** [comment]
@eric-wieser should we plan to do a (video) call some time about this to discuss your thoughts/approach if you have the bandwidth? I would like to make sure we keep moving, and it seems we may have to iterate on it.
If we do, would also invite @shoyer, @njsmith (and probably small group of others).

**Comment by mattip** [comment]
LGTM, there are some long lines >80 columns. I would suggest merging as-is, and opening a new PR as needed for further enhancements.

File: doc/neps/nep-0041-improved-dtype-support.rst [hunk 4]
```diff
@@ -156,6 +159,13 @@ Further, the public API will be designed in a way that is extensible in the futu
    as much as possible. The public API should be an identical, but limited,
    version of the C-API used for the internal NumPy datatypes.
 
+The datatype system may be targeted to work with NumPy arrays,
+for example by providing strided-loops, but should avoid direct interactions with the array-object (typically `np.ndarray` instances).
```
Comment (seberg) [inline doc/neps/nep-0041-improved-dtype-support.rst:163]:
```suggestion
for example by providing strided-loops, but should avoid direct
interactions with the array-object (typically `np.ndarray` instances).
```

**Comment by seberg** [comment]
I will just merge this (broke the long line) then. Thanks everyone.

My offer of a small video chat still stands though, and I would still like to do this if it helps push NEP 41 over the "accepting" finish line. Or alternatively, just as a Q&A style to quickly brief those who only semi-followed things until now.

**Pull request merged**